src/solver/pdist/backward_pdist.cpp

```diff
@@ -0,0 +1,284 @@
+    auto dinput_dtype = miopen::GetDataType(problem.GetdInputDesc().GetType());
+
+    auto dinput_numel = problem.GetdInputDesc().GetElementSize();
+
+    auto input_lengths  = problem.GetInputDesc().GetLengths();
+    auto output_lengths = problem.GetOutputDesc().GetLengths();
+    auto dinput_dims    = problem.GetdInputDesc().GetLengths();
+
+    auto N  = input_lengths[0];
+    auto NO = N * (N - 1) / 2;
+    auto M  = input_lengths[1];
+
+    // Start building result.construction_params
+
+    /* Phrase 1: Calculate gradients for each pair of elements in the input tensor */
+    {
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(NO * M, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        auto kernel        = KernelInfo{};
+        kernel.kernel_file = "MIOpenPdist.cpp";
+        kernel.kernel_name = "PdistBackwardContiguous";
+
+        auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_FP64", static_cast<int>(dtype == miopenDouble)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+        };
+
+        kernel.comp_options = build_params.GenerateFor(kbp::HIP{});
+
+        kernel.l_wk.push_back(xlocalsize);
+        kernel.l_wk.push_back(ylocalsize);
+        kernel.l_wk.push_back(zlocalsize);
+
+        kernel.g_wk.push_back(xgridsize);
+        kernel.g_wk.push_back(ygridsize);
+        kernel.g_wk.push_back(zgridsize);
+
+        result.construction_params.push_back(kernel);
+    }
+
+    /* Phrase 2: Accumulate gradients for each element in the input tensor */
+    // operation: sum
+    // input: ws_dinput (shape=[N-1,N,M])
+    // reduce_dim: 0
+    // output: dinput (shape=[N,M])
+    {
+        // TODO: Add paralellism for efficiency if needed
+        size_t xlocalsize = LOCAL_SIZE;
+        size_t xgridsize  = AlignUp(dinput_numel, xlocalsize);
+        size_t ylocalsize = 1;
+        size_t ygridsize  = 1;
+        size_t zlocalsize = 1;
+        size_t zgridsize  = 1;
+
+        auto kernel = KernelInfo{};
+
+        kernel.kernel_file = "MIOpenReduceCalculation.cpp";
+        kernel.kernel_name = "CalculationFwdContiguous";
+
+        const auto build_params = KernelBuildParameters{
+            {"MIOPEN_USE_FP16", static_cast<int>(dtype == miopenHalf)},
+            {"MIOPEN_USE_FP32", static_cast<int>(dtype == miopenFloat)},
+            {"MIOPEN_USE_BFP16", static_cast<int>(dtype == miopenBFloat16)},
+            {"INPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : input_dtype},
+            {"OUTPUT_TYPE", input_dtype == "bfloat16" ? "ushort" : output_dtype},
```

Comment:
redundant

Reply:
I fixed this.

Pls note here that while I am required to specify both input and output types because I'm using the external pre-defined kernel `CalculationFwdContiguous`, for the `pdist` operation specifically, these types are identical.

Reply:
use dtype for both INPUT_TYPE and OUTPUT_TYPE

Reply:
solved